File: Assets/Scripts/Core/Board/BoardData.cs
using System.Collections.Generic;

namespace Parkiller.Core
{
    public class PlayerLaneData
    {
        public PieceColor Color;
        public int EntryTrackIndex;
        public int HomeEntranceTrackIndex;
        public int CorridorLength;
    }

    // Engine-independent board description consumed by ParchisRules. Produced from a BoardDefinition asset.
    public class BoardData
    {
        public int PlayerCount;
        public int TrackLength;
        public Dictionary<PieceColor, PlayerLaneData> Lanes = new Dictionary<PieceColor, PlayerLaneData>();
        public HashSet<int> SafeTrackIndices = new HashSet<int>();
    }
}

File: src/core/board/boardData.ts
import type { PieceColor } from '../pieceColor'

export interface PlayerLaneData {
  color: PieceColor
  entryTrackIndex: number
  homeEntranceTrackIndex: number
  corridorLength: number
}

// Engine-independent board description consumed by parchisRules. Produced from a BoardDefinition.
export interface BoardData {
  playerCount: number
  trackLength: number
  lanes: Partial<Record<PieceColor, PlayerLaneData>>
  safeTrackIndices: Set<number>
}
